browser/src/App.js

```diff
@@ -58,19 +87,28 @@ function App() {
         Nightfall Phase2 Ceremony
       </div>
       <p>
-        Zero-knowledge proofs require a trusted setup. Since Nightfall uses the Groth16 proving
-        scheme, a second phase of the MPC is needed, for each circuit. We want to invite you to
-        contribute to this Second Phase. To start, just move your {isMobile ? 'device' : 'cursor'}{' '}
-        to generate some entropy. If you want, you can also enter your name for later verification.
-      </p>
-      <p>The process takes 10-20mins. Go grab a coffee!</p>
-      <ContributeCard
-        setEntropy={setEntropy}
-        entropy={entropy}
-        entropyArr={entropyArr}
-        circuits={CIRCUITS}
-        isMobile={isMobile}
-      />
+          Zero-knowledge proofs require a trusted setup. Since Nightfall uses the Groth16 proving
+          scheme, a second phase of the MPC is needed, for each circuit. We want to invite you to
+          contribute to this Second Phase. To start, just move your {isMobile ? 'device' : 'cursor'}{' '}
+          to generate some entropy. If you want, you can also enter your name for later verification.
+        </p>
+      <div style={{display: token == '' ? 'none' : 'inline'}}>
+        <p>The process might takes 10-20mins. Go grab a coffee!</p>
+        <ContributeCard
+          setEntropy={setEntropy}
+          entropy={entropy}
+          entropyArr={entropyArr}
+          circuits={CIRCUITS}
+          isMobile={isMobile}
+          token={token}
+          backendServer={BACKEND_HOST}
+        />
+      </div>
+      <div className="buttons" style={{display: !token ? 'inline' : 'none'}}>
```

Comment:
Same pattern applies here.

Reply:
I replied above, let me know if it makes sense.